### PR TITLE
Updated mariadb to serve over tls

### PR DESCRIPTION
**Pull request by VaniHaripriya**
## The issue resolved by this Pull Request:
Resolves #[RHOAIENG-4972](https://issues.redhat.com//browse/RHOAIENG-4972)

## Description of your changes:

- Updated MariaDB config to server over TLS.
- Added the oc annotation to generate the certs.
- Added a config map with the ssl info.

## Testing instructions

- Deploy DSPO and the dspa using the below yaml file for two scenarios. After deployment, verify that the health 
checks are successful.
- Create a pipeline run and ensure that it completes successfully.
- Execute the following command inside the MariaDB pod terminal to verify that the TLS setup is correctly configured.
 ```
 mysql -u $MYSQL_USER -p$MYSQL_PASSWORD -h 127.0.0.1 -D $MYSQL_DATABASE
    
 MariaDB [mlpipeline]> SHOW GLOBAL VARIABLES LIKE 'have_ssl';
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| have_ssl      | YES   |
+---------------+-------+
1 row in set (0.001 sec)
```
<details>
  <summary>TLS Enabled</summary>

1. Deploy the following DSPA
```
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: sample
spec:
  podToPodTLS: true
  dspVersion: v2
  apiServer:
    enableSamplePipeline: true
    cABundle:
      configMapKey: ca.crt
      configMapName: kube-root-ca.crt
  objectStorage:   
    enableExternalRoute: true
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'  
  mlpipelineUI:
    image: quay.io/opendatahub/ds-pipelines-frontend:latest
```
</details>

<details>
  <summary>TLS Disabled</summary>

1. Deploy the following DSPA
```
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: sample
spec:
  podToPodTLS: false
  dspVersion: v2
  apiServer:
    enableSamplePipeline: true
    cABundle:
      configMapKey: ca.crt
      configMapName: kube-root-ca.crt
  objectStorage:  
    enableExternalRoute: true
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'  
  mlpipelineUI:
    image: quay.io/opendatahub/ds-pipelines-frontend:latest
```
</details>

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-696`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/696/head
git checkout -b pullrequest f3a81ee0eba65788cceed4f723428d6946594abf
oc new-project opendatahub
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-696"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-696`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-696`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-696`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-696`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-696`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-696`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-696`

File: controllers/dspipeline_params.go [hunk 1]
```diff
@@ -89,8 +89,8 @@ type DSPAParams struct {
 	CustomCABundle *dspa.CABundle
 	DSPONamespace  string
 	// Use to enable tls communication between component pods.
-	PodToPodTLS bool
-
+	PodToPodTLS             bool
+	CustomExtraParams       string
```
Comment (hbelmiro) [inline controllers/dspipeline_params.go:93]: It's not 100% clear to me, but I believe TLS should be enabled when `PodToPodTLS` is `true`. Can you confirm @HumairAK?

Comment (gregsheremeta) [reply controllers/dspipeline_params.go:93]: agreed.

so instead of doing this:
```
  database:
    customExtraParams: '{"tls":"true"}'
```

just make that happen automatically when `PodToPodTLS` is `true`

Comment (gregsheremeta) [reply controllers/dspipeline_params.go:93]: also it doesn't look like the value of `customExtraParams` is actually used anywhere, which is a bit puzzling

File: config/internal/mariadb/default/service.yaml.tmpl [hunk 2]
```diff
@@ -3,6 +3,8 @@ kind: Service
 metadata:
   name: mariadb-{{.Name}}
   namespace: {{.Namespace}}
+  annotations:
+    service.beta.openshift.io/serving-cert-secret-name: ds-pipelines-mariadb-proxy-tls-{{.Name}}
```
Comment (gregsheremeta) [inline config/internal/mariadb/default/service.yaml.tmpl:7]: why is this named `mariadb-proxy-tls`? What's the proxy?

File: controllers/dspipeline_params.go (hunk 1 above)
Comment (VaniHaripriya) [reply controllers/dspipeline_params.go:93]: @HumairAK  could you confirm that TLS should be enabled when PodToPodTLS is true. I remember that we discussed to use CustomExtraParams instead of PodToPodTLS in this case, so just want to make sure of that.

Comment (HumairAK) [reply controllers/dspipeline_params.go:93]: sorry for the delayed response, yes that's correct, we only want to enable mariadb when podtopodtls is set to true, in this scenario we should probably configure extra params automatically as needed to make this work

Comment (gregsheremeta) [reply controllers/dspipeline_params.go:93]: > in this scenario we should probably configure extra params automatically as needed to make this work

so we don't need the `customExtraParams` at all. Cool. Thanks for confirming.

File: controllers/database.go [hunk 3]
```diff
@@ -27,13 +27,14 @@ import (
 
 	"errors"
 
+	"os"
```
Comment (gregsheremeta) [inline controllers/database.go:30]: I try to leave stuff like this out of the commit, since it's not part of the change. Usually I do a final pass with "meld" before pushing and undo any stuff like this that my IDE did.

File: controllers/dspipeline_params.go [hunk 4]
```diff
@@ -89,8 +89,7 @@ type DSPAParams struct {
 	CustomCABundle *dspa.CABundle
 	DSPONamespace  string
 	// Use to enable tls communication between component pods.
-	PodToPodTLS bool
-
+	PodToPodTLS             bool
```
Comment (gregsheremeta) [inline controllers/dspipeline_params.go:92]: similarly to my above comment, I would undo this.

**Comment by gregsheremeta** [comment]
some nitpicks, but
/lgtm

**Comment by gregsheremeta** [comment]
/lgtm

nice work!

**Comment by HumairAK** [comment]
awesome thanks folks!

/approve
